sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_models.py

```diff
@@ -2,11 +2,176 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-from typing import TYPE_CHECKING
+from typing import cast, Optional, Union, TYPE_CHECKING
+
+from cryptography.hazmat.primitives.asymmetric.padding import AsymmetricPadding, OAEP, PKCS1v15, PSS, MGF1
+from cryptography.hazmat.primitives.asymmetric.rsa import (
+    RSAPrivateKey,
+    RSAPrivateNumbers,
+    RSAPublicKey,
+    RSAPublicNumbers,
+)
+from cryptography.hazmat.primitives.asymmetric.utils import Prehashed
+from cryptography.hazmat.primitives.hashes import Hash, HashAlgorithm, SHA1, SHA256, SHA384, SHA512
+from cryptography.hazmat.primitives.serialization import Encoding, KeySerializationEncryption, PrivateFormat
+
+from ._enums import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
 
 if TYPE_CHECKING:
-    from . import EncryptionAlgorithm, KeyWrapAlgorithm, SignatureAlgorithm
-    from typing import Any, Optional
+    # Import clients only during TYPE_CHECKING to avoid circular dependency
+    from ._client import CryptographyClient
+    from .._client import KeyClient
+
+
+SIGN_ALGORITHM_MAP = {
+    SHA256: SignatureAlgorithm.rs256,
+    SHA384: SignatureAlgorithm.rs384,
+    SHA512: SignatureAlgorithm.rs512,
+}
+OAEP_MAP = {
+    SHA1: EncryptionAlgorithm.rsa_oaep,
+    SHA256: EncryptionAlgorithm.rsa_oaep_256
+}
+PSS_MAP = {
+    SignatureAlgorithm.rs256: SignatureAlgorithm.ps256,
+    SignatureAlgorithm.rs384: SignatureAlgorithm.ps384,
+    SignatureAlgorithm.rs512: SignatureAlgorithm.ps512,
+}
+
+
+class ManagedRsaKey(RSAPrivateKey):
```

Comment:
Like in .NET, I recommend putting `KeyVault` in the name so it's obvious it's tied to KV (or MHSM) e.g., `KeyVaultRSAKey`. And don't you capitalize acronyms like "RSA" in Python?

Reply:
We don't capitalize "RSA" fully in other classes in the library, but we also don't expose any of those types. Here's an example: https://github.com/Azure/azure-sdk-for-python/blob/ecbb63988a039e0d73eedcdb55f8ad6873e315f4/sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_internal/rsa_key.py#L25

Since "RSA" is used for the interface name, it implies that we could do the same here. @annatisch, what do you think? Would something like `KeyVaultPrivateRSAKey` (to match an incoming `KeyVaultPublicRSAKey`) work? Or should we trim this down somewhere?

Reply:
Yup I think those two names are good - though should "RSA" come first to match the cryptography objects?
i.e. `KeyVaultRSAPrivateKey` and `KeyVaultRSAPublicKey`?

Reply:
FWIW, that's basically what I did in .NET: used the naming convention of .NET's crypto primitives and their implementations in my own naming.